mne/encoding/feature.py

```diff
@@ -0,0 +1,298 @@
+"""Classes for transforming, preprocessing, and fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Creates delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.0
+    """
+    def __init__(self, delays=None, sfreq=1.):
+        delays, sfreq = _check_delayer_params(
+            delays, sfreq)
+        self.delays = delays
+        self.sfreq = float(sfreq)
+
+    def fit(self, X, y=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+        y : array
+
+        Returns
+        -------
+        self
+        """
+        return self
+
+    def transform(self, X=None, y=None):
+        """Create a time-lagged representation of X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_times, n_features)
+            The input data to be time-lagged and returned.
+
+        Returns
+        -------
+        X_delayed : array, shape (n_times, n_feature * n_delays)
+            The delayed featuers of X.
+        """
+        X_delayed, names = delay_timeseries(X, delays=self.delays,
+                                            sfreq=self.sfreq)
+        self.delays_ = names
```

Comment:
Not strictly necessary but it can be complicated to keep track of which output columns map to which delay/feature pairs, so that's what this is for. I think I'll rename it to `names_` because I think `delays_` is confusing